src/doc/trpl/closures.md

```diff
@@ -1,214 +1,510 @@
 % Closures
 
-So far, we've made lots of functions in Rust, but we've given them all names.
-Rust also allows us to create anonymous functions. Rust's anonymous
-functions are called *closures*. By themselves, closures aren't all that
-interesting, but when you combine them with functions that take closures as
-arguments, really powerful things are possible.
+Rust has not only named functions, but anonymous functions as well. Anonymous
+functions that have an associated environment are called 'closures,' and Rust
+has a really great implementation of them, as we'll see.
 
-Let's make a closure:
+# Syntax
 
-```{rust}
-let add_one = |x| { 1 + x };
+Closures look like this:
 
-println!("The sum of 5 plus 1 is {}.", add_one(5));
+```rust
+let plus_one = |x| x + 1;
```

Comment:
one warning: I expect that the fix for https://github.com/rust-lang/rust/issues/23319 will cause this not to work unless you annotate the type of `x` (`|x: i32|`)